code/ship/ship.cpp

```diff
@@ -3291,10 +3302,12 @@ int parse_ship_values(ship_info* sip, const bool is_template, const bool first_t
 		Error(LOCATION, "%s '%s' has too many primary banks (%d).  Maximum for ships is currently %d.\n", info_type_name, sip->name, sip->num_primary_banks, MAX_SHIP_PRIMARY_BANKS);
 	}
 
+	memset(sip->allowed_weapons, 0, sizeof(int) * MAX_WEAPON_TYPES);
```

Comment:
I agree that changing all instances is not in the scope of this PR. However, introducing a minor inconsistency here will improve the code quite a bit (IIRC you even mentioned on IRC that you had issues with memsetting too much memory).